nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/util/CertificateUtils.java

```diff
@@ -625,6 +634,20 @@ public static boolean isTlsError(Throwable e) {
         }
     }
 
+    /**
+     *Extracts the common name from the given DN.
+     *
+     * @param dn the distinguished name to evaluate
+     * @return the common name if it exists, null otherwise.
+     */
+    public static String getCommonName(String dn){
```

Comment:
Recommend declaring the parameter as `final`:
```suggestion
    public static String getCommonName(final String dn) {
```